components/interface/TreeWidget.js

```diff
@@ -383,47 +382,97 @@ export default class TreeWidget extends React.Component {
     });
   }
 
+  findRoot (nodes) {
+    let min = nodes[0].id;
+    for ( let i = 1; i < nodes.length; i++) {
+      if(nodes[i].id < min) {
+        min = nodes[i].id;
+      }
+    }
+    return min;
+  }
+
   componentWillMount () {
     if (window.templateID !== undefined) {
       this.initTree(window.templateID);
+      document.addEventListener('mousedown', this.monitorMouseClick, false);
     }
   }
 
+  componentWillReceiveProps () {
+    document.removeEventListener('mousedown', this.monitorMouseClick, false);
+  }
+
   nodeClick (event, rowInfo) {
     console.log("clicked on the tree node");
     console.log(rowInfo);
     this.selectNode(rowInfo.node);
   }
 
+  monitorMouseClick (e) {
+    if (this.colorPickerNode !== undefined && this.colorPickerNode.contains(e.target)) {
+      return;
```

Comment:
@ddelpiano to keep in mind - I tend not to like and generally avoid returns in the middle of a function... I'd rather figure out a condition do the if on that and do nothing and let the function return naturally (here not a big issues because it's a small function but it becomes unreadable quickly if it's a bigger chunk of code)

Reply:
You could easily restructure this code reversing the condition and removing the return altogether and it just looks cleaner